Documentation/architecture.rst

```diff
@@ -208,174 +208,241 @@ label source defaults to ``any:`` which will match all labels regardless of
+tracing back which container labels are assigned to the origin container.
 
 .. note::
 
    As the packet headers contain security sensitive information, it is higly
    recommended to either encrypt all traffic or run Cilium in a trusted network
    environment.
 
-There are two possible approaches to performing network forwarding for
-container-to-container traffic:
+Cilium keeps the networking concept as simple as possible. There are two
+networking models to choose from.
 
-* **Overlay Routing:** In this mode, the network connecting the container
-  hosts together does not need to be aware of the *node prefix* or the IP
-  addresses of containers.  Instead, a *virtual overlay network* is created on
-  top of the existing network infrastructure by creating tunnels between
-  containers hosts using encapsulation protocols such as VXLAN, GRE, or Geneve.
-  This minimizes the requirements on the underlying network infrastructure. The
-  only requirement in this mode is for containers hosts to be able to reach
-  each other by UDP (VXLAN/Geneve) or IP/GRE. As this requirement is typically
-  already met in most environments, this mode usually does not require
-  additional configuration from the user. Cilium can deterministically map from
-  any container IP address to the corresponding node IP address, Cilium can
-  look at the destination IP address of any packet destined to a container, and
-  then use encapsulation to send the packet directly to the IP address of the
-  node running that container. The destination node then decapsulates the
-  packet, and delivers it to the local container.  Because overlay routing
-  requires no configuration changes in the underlying network, it is often the
-  easiest approach to adopt initially.
-
-* **Direct Routing:**  In this mode, Cilium will hand all packets that are not
-  addressed to a local container and not addressed to the local node to the
-  Linux stack causing it to route the packet as it would route any other
-  non-local packet. As a result, the network connecting the Linux node hosts
-  must be aware that each of the node IP prefixes are reachable by using the
-  node's primary IP address as an L3 next hop address.   In the case of a
-  traditional physical network this would typically involve announcing each
-  node prefix as a route using a routing protocol within the datacenter. Cloud
-  providers (e.g, AWS VPC, or GCE Routes) provide APIs to achieve the same result.
+- `Overlay / Encapsulation mode`_ 
+- `Direct / Native Routing Mode`_
 
 Regardless of the option chosen, the container itself has no awareness of the
 underlying network it runs on, it only contains a default route which points to
-the IP address of the container host. Given the removal of the routing cache in
+the IP address of the cluster node Given the removal of the routing cache in
 the Linux kernel, this reduces the amount of state to keep to the per
```

Comment:
amount of state to keep in the per-connection flow cache, which allows for the termination of millions of connection in each container.